### ISSUE TITLE
[Jira Time Tracking] Please add support for On-Prem Jira installation

### ISSUE THREAD

**Issue by marcgemis** [raycast/extensions#12550]
### Extension

https://www.raycast.com/niallpaterson/jira-time-tracking

### Description

The plug-in does not work for an on-prem installation of Jira.
I tried to compare the Jira and Jira On-Prem plug-ins. I think the Authorization headers are slightly different.
Would it possible make the plug-in compatible with an on-prem Jira? Or release a separate version?

### Who will benefit from this feature?

Everyone with an on-prem installation of Jira

**Comment by haydencbarnes** [comment]
Hi @marcgemis would you like to test https://github.com/raycast/extensions/pull/11918 ? There is a [guide here](https://github.com/raycast/extensions/pull/11918#issuecomment-2067870064) on how to do it fast :)

**Comment by marcgemis** [comment]
HI @haydencbarnes,
I tried it, but is seems that our server is responding with HTML instead of JSON when the list of projects is retrieved.
I'm not familiar yet with debugging Raycast extensions, but is there a log that I can send to you?

**Comment by marcgemis** [comment]
In the terminal window I only see
`ready -  built extension successfully
06:47:28.296 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:48:47.693 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:49:30.970 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:50:37.461 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:51:15.934 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:51:38.589 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
06:54:35.124 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
07:04:35.378 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
07:06:56.909 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0
07:07:01.819 Fetching projects from: /rest/api/2/project?maxResults=500&startAt=0`

**Comment by marcgemis** [comment]
I added some log instructions, e.g. in requests.ts, just above the const responseBody = await res.json()

The "res" is

`
07:20:36.988 The response: Response {

  size: 0,
  [Symbol(Body internals)]: {
    body: PassThrough {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 6,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: true,
      [Symbol(kCapture)]: false,
      [Symbol(kCallback)]: null
    },
    stream: PassThrough {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 6,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: true,
      [Symbol(kCapture)]: false,
      [Symbol(kCallback)]: null
    },
    boundary: null,
    disturbed: false,
    error: null
  },
  [Symbol(Response internals)]: {
    type: 'default',
    url: 'https://********/rest/api/2/project?maxResults=500&startAt=0',
    status: 403,
    statusText: '',
    headers: {
      connection: 'close',
      'content-security-policy': "frame-ancestors 'self'",
      'content-type<…>
`

of course the ****** is the name of our jira host. When I click on that URL in the browser I do get the projects in json format.

**Comment by marcgemis** [comment]
When I replace
`

/**  Authorization: `Basic ${Buffer.from(`${prefs.username}:${prefs.token}`).toString("base64")}`, */
   Authorization: `Bearer $2{prefs.token}`,
`
in requests.ts, I get a "Projects loaded (0/1)" in the Raycast window.
The above authorization line is based on the code found in the  jira-search-self-hosted extension.

**Comment by haydencbarnes** [comment]
@marcgemis, that's interesting. As this is not a file I changed, let me look through it here quickly and confirm that it also works for the cloud. It's interesting that it also gives you a 403, and for me, it does not.

**Comment by haydencbarnes** [comment]
Bearer does not work as nicely for the cloud, unfortunately. According to https://developer.atlassian.com/server/jira/platform/basic-authentication/  this current request should work fine, and works for me on multiple instances. @marcgemis Can you please shoot me a message over on https://raycastcommunity.slack.com/team/U06UKMRBUU9 with a screenshot of your config for the extension.

**Comment by haydencbarnes** [comment]
![CleanShot 2024-06-10 at 01 45 27@2x](https://github.com/raycast/extensions/assets/120114860/bfc915f6-bce9-4c09-98ae-31a43332e58b)
These guys are what I am interested in

**Comment by haydencbarnes** [comment]
Also, I would like to confirm that you are using your personal API key and not your organization's API key.

**Comment by haydencbarnes** [comment]
> https://********/rest/api/2/project?maxResults=500&startAt=0'

The URL works because you are already authenticated, the 403 error you are receiving here is a credentials issue.

**Comment by marcgemis** [comment]
> Also, I would like to confirm that you are using your personal API key and not your organization's API key?

Yes, I generated a key in Jira via Profile > Personal Access Tokens

We do have Jira v9.12.7#9120007

I don't understand it anymore. Now it works with both methods of authentication, and with email and user name in the settings. Maybe something is cached now?

**Comment by haydencbarnes** [comment]
Hmmm, you could try restarting raycast or deleting the branch and rebuilding it to see if it works now with a reset cache.

**Comment by marcgemis** [comment]
After stopping Raycast, removing the plugin folder, restarting Raycart and retrieving the plugin again, it stopped working.
I tried again with email and user name. Nothing worked.
As soon as I use the authentication with the Bearer token, it works.

**Comment by haydencbarnes** [comment]
@marcgemis Copy that. Since I have already written the code for switching between APIs for Server/Cloud, will submit a PR here in a minute that switches between bearer for Server and basic for Cloud.

**Comment by haydencbarnes** [comment]
@marcgemis Can you please pull the latest and retry? The update I just submitted to requests.ts should fix your issue.

**Comment by marcgemis** [comment]
seems to work for the project I tried yesterday. (ca 3000 issues)
I seem to have problems with a project with + 23000 issues (+10 year old project) but that is unrelated to this.

**Comment by haydencbarnes** [comment]
> seems to work for the project I tried yesterday. (ca 3000 issues)
> 
> I seem to have problems with a project with + 23000 issues (+10 year old project) but that is unrelated to this.

How many of the 23,000 does it load?

**Comment by haydencbarnes** [comment]
Nevermind, I see https://github.com/raycast/extensions/issues/12886

**Comment by haydencbarnes** [comment]
@pernielsentikaer This can be closed courtesy of  https://github.com/raycast/extensions/pull/11918

**Comment by haydencbarnes** [comment]
@raycastbot close this issue
